docs/source/extensions/datamodules.rst

```diff
@@ -195,16 +196,17 @@ settings.
             MNIST(os.getcwd(), train=False, download=True, transform=transforms.ToTensor())
 
 
-.. warning:: ``prepare_data`` is called from a single process (e.g. GPU 0). Do not use it to assign state (`self.x = y`).
+.. warning:: ``prepare_data`` is called from a single process across all nodes. It is not recommended to assign state here (e.g. ``self.x = y``).
```

Comment:
can someone confirm this?

Reply:
Depends on the value of `LightningModule.prepare_data_per_node`:

https://github.com/PyTorchLightning/pytorch-lightning/blob/2036dfb5df2a48056d1c043ee311aed793187070/pytorch_lightning/trainer/connectors/data_connector.py#L159-L160